addons/account_edi/models/account_move.py

```diff
@@ -226,35 +226,75 @@ def _serialize_python_dictionary(vals):
+                    'tax_id': tax,
+                    'src_tax_id': src_tax,
+                    'tax_repartition_line_id': tax_line.tax_repartition_line_id,
+                    'base_amount': row['base_amount'],
+                    'tax_amount': row['tax_amount'],
+                    'base_amount_currency': row['base_amount_currency'],
+                    'tax_amount_currency': row['tax_amount_currency'],
+                })
+            return invoice_lines_tax_values_dict
+
+        def compute_invoice_lines_tax_values_dict_from_compute_all(invoice_lines):
+            invoice_lines_tax_values_dict = {}
+            sign = -1 if self.is_inbound() else 1
+            for invoice_line in invoice_lines:
+                taxes_res = invoice_line.tax_ids.compute_all(
+                    invoice_line.price_unit * (1 - (invoice_line.discount / 100.0)),
+                    currency=invoice_line.currency_id,
+                    quantity=invoice_line.quantity,
+                    product=invoice_line.product_id,
+                    partner=invoice_line.partner_id,
+                    is_refund=invoice_line.move_id.move_type in ('in_refund', 'out_refund'),
+                )
+                invoice_lines_tax_values_dict[invoice_line] = []
+                rate = abs(invoice_line.balance) / abs(invoice_line.amount_currency) if invoice_line.amount_currency else 0.0
+                for tax_res in taxes_res['taxes']:
+                    invoice_lines_tax_values_dict[invoice_line].append({
+                        'base_line_id': invoice_line,
+                        'tax_id': self.env['account.tax'].browse(tax_res['id']),
+                        'tax_repartition_line_id': self.env['account.tax.repartition.line'].browse(tax_res['tax_repartition_line_id']),
+                        'base_amount': sign * invoice_line.company_currency_id.round(tax_res['base'] * rate),
+                        'tax_amount': sign * invoice_line.company_currency_id.round(tax_res['amount'] * rate),
+                        'base_amount_currency': sign * tax_res['base'],
+                        'tax_amount_currency': sign * tax_res['amount'],
+                    })
+            return invoice_lines_tax_values_dict
+
+        # Decide whether we will use AccountTax.compute_all or AccountMoveLine._get_query_tax_details to compute
+        # the tax amounts for each invoice.
+        # By default: if using 'Round Locally', we use compute_all (which will give rounded amounts for each line)
+        #             if using 'Round Globally', we use _get_query_tax_details (which will allocate the total tax amount proportionately among the lines)
+        # This can be changed using the 'use_sql_query_tax_details' context key.
+        company = self.company_id or self.env.company_id
+        use_sql_query_tax_details = bool(self.env.context.get('use_sql_query_tax_details', company.tax_calculation_rounding_method == 'round_globally'))
```

Comment:
add an explicit parameter. This issue is not related to the `round_globally` method

Reply:
@smetl actually, the new 15.0 method causes issues only when the rounding method is set to 'Round per Line'. AFAIK it works fine with 'Round Globally'. 

Conversely, the 14.0 method didn't handle well 'Round Globally'.

(see https://www.odoo.com/web#id=2593016&cids=1&menu_id=4720&action=333&active_id=49&model=project.task&view_type=form)

So doesn't it makes sense to keep the 15.0 method when the user uses 'Round Globally', and use the 14.0 method when the user uses 'Round Locally'?

Reply:
Happy to change the context key into an explicit parameter though if you feel that's better :smile:

Reply:
Done :+1: